core/src/net/sf/openrocket/masscalc/MassCalculation.java

```diff
@@ -273,9 +289,16 @@ MassCalculation calculateStructure() {
 		// if( this.config.isComponentActive(component) ){
 		// 	System.err.println(String.format( "%s>>[%s]....", prefix, component.getName()));
 		// }
-		
+
+		if(null != analysisMap) {
+			if (this.config.isComponentActive(component) && (! analysisMap.containsKey(component.getName()))){
+				CMAnalysisEntry entry = new CMAnalysisEntry(component);
+				analysisMap.put(component.getName(), entry);
```

Comment:
The component.getName() is not sufficiently unique, I think you'll need component.getID(). Multiple components can have the same name (in fact, the default name for components is chosen by default). Come to think of it, I wonder if we have this problem elsewhere.

Reply:
Yea, I'd tried that, but it broke.  

But, It's still a valid point.  I'll convert that over.

Reply:
seems to work.   marking as resolved.